### PR TITLE
Fix middleware build result. Return middleware next when invalid result.

### PR DESCRIPTION
**Pull request by gdborton**
### Related Issues

This fixes an issue in canary builds where if the user's middleware doesn't return anything, the request errors.

Now we'll return a response that indicates that we've not handled the request.

#### Tests

- [x] The code changed/added as part of this PR has been covered with tests
- [x] All tests pass locally with `yarn test-unit`

#### Code Review

- [x] This PR has a concise title and thorough description useful to a reviewer
- [ ] Issue from task tracker has a link to this PR

### CODE REVIEW

**Review by styfle — COMMENTED** [review]
Can you add a test to ensure this is working?

File: packages/middleware/src/entries.js [hunk 1]
```diff
@@ -10,6 +10,15 @@ _ENTRIES['middleware_pages/_middleware'] = {
         response: result,
       };
     }
-    return result;
+
+    return {
+      promise: Promise.resolve(),
+      waitUntil: Promise.resolve(),
+      response: new Response(null, {
```
Comment (gdborton) [inline packages/middleware/src/entries.js]: @AndyBitz is this assumption correct? Whenever middleware returns something that isn't a `Response` we essentially call next?

File: packages/middleware/src/entries.js [hunk 2]
```diff
@@ -10,6 +10,15 @@ _ENTRIES['middleware_pages/_middleware'] = {
         response: result,
       };
     }
-    return result;
+
```
Comment (styfle) [inline packages/middleware/src/entries.js]: What about if the user does `return null`? Do we handle that differently than `return undefined`?

Comment (gdborton) [reply packages/middleware/src/entries.js]: The platform always expects a result like `{ waitUntil, response, promise }` I'm not aware of any special convention for returning null from a middleware.

File: packages/middleware/src/entries.js (hunk 1 above)
Comment (AndyBitz) [reply packages/middleware/src/entries.js]: Yes, but we should check what Next.js does to make sure we don't create any differences in that behaviour

**Comment by gdborton** [comment]
> Can you add a test to ensure this is working?

Yep!, but whether or not it "works" depends mostly on whether the result matches what the platform expects.

File: packages/middleware/test/build.test.ts [hunk 3]
```diff
@@ -17,8 +25,30 @@ describe('build()', () => {
...
+    expect(String(handledResponse.response.body)).toEqual('Hi from the edge!');
+    expect(
+      (handledResponse.response as Response).headers.get('x-middleware-next')
+    ).toEqual(null);
+    expect(unhandledResponse.response.body).toEqual(null);
     expect(
-      await fsp.stat(join(fixture, '.output/server/pages/_middleware.js'))
-    ).toBeTruthy();
+      (unhandledResponse.response as Response).headers.get('x-middleware-next')
+    ).toEqual('1');
```
Comment (gdborton) [inline packages/middleware/test/build.test.ts:55]: @styfle Updated w/ a unit test that ensures returning nothing from userland middleware gets wrapped with a response w/ `x-middleware-next` set.

**Review by styfle — APPROVED** [review]
Great work, thanks! 🎉

File: packages/middleware/src/entries.js (hunk 1 above)
Comment (gdborton) [reply packages/middleware/src/entries.js]: @AndyBitz I checked next.js and here's what they're doing - https://github.com/vercel/next.js/blob/canary/packages/next/server/web/adapter.ts#L35

No checking for the result to be an instance of Response, if it's truthy it's returned. I've updated this PR to do the same.

**Review by AndyBitz — APPROVED** [review]

**Pull request merged**